src/Clock.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+
+type State = {
+  today: Date,
+};
+
+type Props = {
+  clockName: string,
+};
+
+export class Clock extends Component<Props, State> {
+  newTime = 0;
+
+  state: Readonly<State> = {
+    today: new Date(),
+  };
+
+  componentDidMount(): void {
+    this.newTime = window.setInterval(() => {
+      this.setState({ today: new Date() });
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State): void {
+    if (prevState.today !== this.state.today) {
+      // eslint-disable-next-line no-console
+      console.info(this.state.today.toUTCString().slice(-12, -4));
+    }
+
+    if (prevProps.clockName !== this.props.clockName) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevProps.clockName} to ${this.props.clockName}`);
+    }
+  }
+
+  componentWillUnmount(): void {
+    window.clearInterval(this.newTime);
+  }
+
+  render(): React.ReactNode {
+    const { today } = this.state;
+    const { clockName } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {today.toUTCString().slice(-12, -4)}
```

Comment:
DRY, looks like a new function to create 😉

Reply:
done!)